### ISSUE TITLE
detach

### ISSUE THREAD

**Issue by zhanghongruiupup** [MalongTech/research-xbm#6]
在do_train里必须用detach截断feats和targets吗？

**Comment by zhanghongruiupup** [comment]
你好请问下相关细节，打扰了。   loss = loss + cfg.XBM.WEIGHT * xbm_loss，这样就是2倍batch的数据在做对比loss，为什么不直接loss=xbm_loss了

**Comment by bnu-wangxun** [comment]
1. 是的，需要detach. 

2. loss当前的mini-batch也很重要，所以不是只用xbm_loss。也可以先把当前mini-batch enqueue 到xbm,效果差不多

**Comment by wangxinrugithub** [comment]
> 1. 是的，需要detach.
> 2. loss当前的mini-batch也很重要，所以不是只用xbm_loss。也可以先把当前mini-batch enqueue 到xbm,效果差不多

针对你说的2 我仔细看了代码  当前代码是先把当前的mini_batch enqueue到xbm里面，之后才计算的loss

**Comment by wangxinrugithub** [comment]
> 1. 是的，需要detach.
> 2. loss当前的mini-batch也很重要，所以不是只用xbm_loss。也可以先把当前mini-batch enqueue 到xbm,效果差不多

我只使用xbm_loss。出现了loss不收敛的情况
